### DISCUSSION TITLE
Missing ical_uid after new outlook event .save

### DISCUSSION THREAD

**Discussion by ghost** [O365/python-o365#896] [General]
Greetings, 
A question about saving a new outlook event using MS Graph through O365. 

After new_outlook_event.save various properties of the new_outlook_event object are updated. However my much needed ical_uid (to uniquelly identify an event) has no value.

How can I get the ical_uid value just after creating a new event? Did I miss something considering the .save method?

**Comment by alejcas** [comment]
Can you tell where this "ical_uid" is in the ms graph docs?

**Comment by ghost** [comment]
ical_uid is one of the many properties of an outlook event

[image: ical_uid_1.PNG]
Here the value is None (just after creating a new event with
outllok_event.save()


[image: ical_uid_2.PNG]

Here the value is the unique identifier of an event when reading it back

See also: event resource type - Microsoft Graph v1.0 | Microsoft Learn
<https://learn.microsoft.com/en-us/graph/api/resources/event?view=graph-rest-1.0>

[image: image.png]

**Reply by alejcas** [reply]
I see. Changed this in [5a9450a](https://github.com/O365/python-o365/commit/5a9450af39f9e470759a74c68cde2f257ca662c7)

**Reply by alejcas** [reply]
Will release a new version with this change soon. For the moment you can overwrite the event.save method

**Comment by ghost** [comment]
Thank you! I was about to have a look at the source code anyway. Eager to
learn...
